openmc/deplete/results_list.py

```diff
@@ -192,3 +193,72 @@ def get_depletion_time(self):
         for ix, res in enumerate(items):
             times[ix] = res.proc_time
         return times
+
+    def export_to_materials_xml(self, burnup_index, nuc_with_data=None, new_xml_name=None):
+        """Export Materials.xml file from the depletion results
+
+        Parameters
+        ----------
+        burn_index : int
+            Index of burnup step to evaluate
+        nuc_with_data : Iterable of str, optional
+            nuclides list to evaluate with neutron data. If not provided,
+            nuclides from the cross_sections element of materials.xml will
+            be used. If that element is not present, nuclides from
+            OPENMC_CROSS_SECTIONS will be used.
+        new_xml_name : str, optional
+            name of xml file to put depleted materials into. If not
+            provided, no new xml file will be written.
+
+        Returns
+        -------
+        mat_file : Materials
+            A modified Materials instance containing depleted material data,
+            plus original isotopic compositions of non-depletable materials
+        """
+        result = self[burnup_index]
+
+        # Only materials found in the original materials.xml file will be
+        # updated. If for some reason you have modified OpenMC to produce
+        # new materials as depletion takes place, this method will not
+        # work as expected and leave out that material.
+        mat_file = openmc.Materials.from_xml("materials.xml")
+
+        # Only nuclides with valid transport data will be written to
+        # the new materials XML file. The precedence of nuclides to select
+        # is first ones provided as a kwarg here, then ones specified
+        # in the materials.xml file if provided, then finally from
+        # the environment variable OPENMC_CROSS_SECTIONS.
+        if nuc_with_data:
+            available_cross_sections = nuc_with_data
+        else:
+            # select cross_sections.xml file to use
+            if mat_file.cross_sections:
+                this_library = DataLibrary.from_xml(path=mat_file.cross_sections)
+            else:
+                this_library = DataLibrary.from_xml()
+
+            # Find neutron libraries we have access to
+            available_cross_sections = []
+            for lib in this_library.libraries:
+                if lib['type'] == 'neutron':
+                    available_cross_sections.extend(lib['materials'])
+            if not available_cross_sections:
+                raise Exception('No neutron libraries found in cross_sections.xml')
```

Comment:
Not a huge fan of throwing a generic `Exception` but I also can't think of a great replacement. Maybe something like https://github.com/openmc-dev/openmc/blob/2115cae40aa242ec0e2df31c8a43b92f2ba3babb/openmc/exceptions.py#L21

Reply:
Why add this extra code when it would convey precisely the same message? I find it unlikely here other code would wrap this in sufficiently complicated exception handling blocks to where throwing a plain Exception would be inconvenient.

Reply:
The net effect to someone who isn't wrapping this in try / except blocks doesn't change, absolutely. But this also helps distinguish between exceptions that may be raised through this routine however small it may be.

The reason I would prefer raising a `DataError` is that the type of exception raised more closely relates to why the exception was raised. And the extra code would just be an import at the top of the file and switching this line

Reply:
Ah, didn't realize the data error exception was already present in OpenMC. I was kind of tired while responding to feedback earlier. Will definitely add this on the next iteration.

Reply:
Alright, got this in.